main/main.cpp

```diff
@@ -1,13 +1,28 @@
+#include <unistd.h>
+
 #include <block.hpp>
 #include <iostream>
 #include <memory>
+#include <thread>
+#include <chrono>
 #include <vector>
 
+#include "board.hpp"
+#include "cli_color.hpp"
 #include "diagonal.hpp"
+#include "drawable.hpp"
```

Comment:
thats a lot of includes :P You should create shape factory and only include it :P

Reply:
It was done in the other PR but i didn't rebase yet